mempool/src/shared_mempool/coordinator.rs

```diff
@@ -75,128 +74,22 @@ pub(crate) async fn coordinator<V>(
-                                let timeline_state = match peer_manager
-                                    .is_upstream_peer(&peer, None)
-                                {
-                                    true => TimelineState::NonQualified,
-                                    false => TimelineState::NotReady,
-                                };
-                                // This timer measures how long it took for the bounded executor to
-                                // *schedule* the task.
-                                let _timer = counters::TASK_SPAWN_LATENCY
-                                    .with_label_values(&[counters::PEER_BROADCAST_EVENT_LABEL, counters::SPAWN_LABEL])
-                                    .start_timer();
-                                // This timer measures how long it took for the task to go from scheduled
-                                // to started.
-                                let task_start_timer = counters::TASK_SPAWN_LATENCY
-                                    .with_label_values(&[counters::PEER_BROADCAST_EVENT_LABEL, counters::START_LABEL])
-                                    .start_timer();
-                                bounded_executor
-                                    .spawn(tasks::process_transaction_broadcast(
-                                        smp_clone,
-                                        transactions,
-                                        request_id,
-                                        timeline_state,
-                                        peer,
-                                        task_start_timer
-                                    ))
-                                    .await;
-                            }
-                            MempoolSyncMsg::BroadcastTransactionsResponse{request_id, retry, backoff} => {
-                                let ack_timestamp = SystemTime::now();
-                                peer_manager.process_broadcast_ack(PeerNetworkId(network_id.clone(), peer_id), request_id, retry, backoff, ack_timestamp);
-                            }
-                        }
-                    }
-                    Event::RpcRequest(peer_id, _msg, _res_tx) => {
-                        counters::UNEXPECTED_NETWORK_MSG_COUNT
-                            .with_label_values(&[&network_id.network_id().to_string(), &peer_id.to_string()])
-                            .inc();
-                        sample!(
-                            SampleRate::Duration(Duration::from_secs(60)),
-                            warn!(
-                                LogSchema::new(LogEntry::UnexpectedNetworkMsg)
-                                    .peer(&PeerNetworkId(network_id, peer_id))
-                            )
-                        );
-                    }
-                }
+                handle_event(&executor, &bounded_executor, &mut scheduled_broadcasts, &mut smp, network_id, event).await;
```

Comment:
handle network event? also you probably noticed this, but this isn't actually the network interface, it is some extra thing on top of it. (I think)